physx/src/query.rs

```diff
@@ -40,251 +38,374 @@ impl Into<physx_sys::PxQueryFilterData> for PxQueryFilterData {
     }
 }
 
-#[repr(transparent)]
-pub struct PxRaycastCallback {
-    pub obj: Owner<physx_sys::PxRaycastCallback>,
-}
+// RAYCAST CALLBACKS
 
-#[repr(transparent)]
-pub struct PxRaycastBuffer(PxRaycastCallback);
+pub struct RaycastCallbackDefault;
```

Comment:
The use-case here should be filled by a `PxRaycastBuffer` wrapper instead.  There should be a `RaycastCallback` trait for the bound used in `Scene::query` for consistency with the class-as-trait pattern.  However, I'm not sure if the current `RaycastCallback` trait will work right now, you may need `Class<physx_sys::PxRaycastCallback>` and maybe `UserData` (see the other note about that, `PxRaycastBuffer` would define it as `()`) as super traits.

Also, not sure whether using `*Callback` as the name for user defined trait overrides or `*/Px*` as class `trait/struct` names should take precedence here. It might be possible to overlap the concerns, but it's kinda confusing me trying to describe it here so perhaps they should get split.